internal/compiler/program.go

```diff
@@ -454,7 +454,7 @@ func (p *Program) getSemanticDiagnosticsForFile(ctx context.Context, sourceFile
 
 	isPlainJS := ast.IsPlainJSFile(sourceFile, compilerOptions.CheckJs)
 	if isPlainJS {
-		diags = core.Filter(diags, func(d *ast.Diagnostic) bool {
+		return core.Filter(diags, func(d *ast.Diagnostic) bool {
```

Comment:
may be add !!! here so we know we need to redo the work ?

Reply:
Left a TODO, but need a reapproval.